### ISSUE TITLE
Analyzer throws unexpected compile error when checks least and greatest closures for invariant generic typedef

### ISSUE THREAD

**Issue by iarkh** [dart-lang/sdk#44164]
Dart SDK version: 2.12.0-29.0.dev (dev) (Mon Nov 9 06:42:00 2020 -0800) on "windows_x64"

This issue is similar with #44163, but it's against the analyzer whereas #44163 is about dart.

Here is a source code example:

``` dart
X captureTypeArgument<X>() => throw "Error";

typedef check<X> = X Function(X);

void main() {
  void f<X>(check<X> Function() g) => g();
  f(() => captureTypeArgument());
}
```
Seems like analyzer should pass and dart should throws a runtime error here. However, this is not so and compile error appears instead.

Analyzer sample output is:
> Analyzing test.dart...
>   error - Couldn't infer type parameter 'X'.
> 
> Tried to infer 'Object?' for 'X' which doesn't work:
>   Parameter 'g' declared as     'X Function(X) Function()'
>                 but argument is 'Object? Function(Never) Function()'.
> 
> Consider passing explicit type argument(s) to the generic.
> 
>  - test.dart:7:3 - could_not_infer
>   error - The argument type 'Object? Function(Never) Function()' can't be assigned to the parameter type 'Object? Function(Object?) Function()'. - test.dart:7:5 - argument_type_not_assignable
> 2 errors found.

**Comment by iarkh** [comment]
Seems like this issue should be closed according to the #44163 evaluation.
@eernstg, could you please confirm?

**Comment by eernstg** [comment]
Yes, we get the greatest closure `Object? Function(Never)` here again, and there is no type argument to `f` which could make `f(() => captureTypeArgument())` type correct. So the analyzer behavior is correct and we should close this issue.

Thanks for spotting this!
